packages/compiler/test/config/config-interpolation.test.ts

```diff
@@ -0,0 +1,316 @@
+          },
+        },
+      });
+    });
+
+    it("emitter options can interpolate each other", () => {
+      const config = {
+        ...defaultConfig,
+        projectRoot: "/dev/ws",
+        outputDir: "/my-custom-output-dir",
+        emitters: {
+          emitter1: {
+            "emitter-output-dir": "{output-dir}/{emitter-folder}",
+            "emitter-folder": "custom-1",
+          },
+        },
+      };
+      const resolved = expectExpandConfigVariables(config, { cwd: "/dev/wd" });
+      deepStrictEqual(resolved, {
+        ...config,
+        emitters: {
+          emitter1: {
+            "emitter-output-dir": "/my-custom-output-dir/custom-1",
+            "emitter-folder": "custom-1",
+          },
+        },
+      });
+    });
+  });
+
+  describe("validateConfigPathsAbsolute", () => {
+    it("emit diagnostic for using a relative path starting with ./", () => {
+      const config = {
+        ...defaultConfig,
+        projectRoot: "/dev/ws",
+        outputDir: "./my-output",
+      };
+      const diagnostics = validateConfigPathsAbsolute(config);
+      expectDiagnostics(diagnostics, {
+        code: "config-path-absolute",
+        message: `Path "./my-output" cannot be relative. Use {cwd} or {project-root} to specify what the path should be relative to.`,
+      });
+    });
+
+    it("emit diagnostic for using a relative path starting with ../", () => {
+      const config = {
+        ...defaultConfig,
+        projectRoot: "/dev/ws",
+        outputDir: "../my-output",
+      };
+      const diagnostics = validateConfigPathsAbsolute(config);
+      expectDiagnostics(diagnostics, {
+        code: "config-path-absolute",
+        message: `Path "../my-output" cannot be relative. Use {cwd} or {project-root} to specify what the path should be relative to.`,
+      });
+    });
+
+    it("emit diagnostic for using a relative path", () => {
+      const config = {
+        ...defaultConfig,
+        projectRoot: "/dev/ws",
+        outputDir: "my-output",
+      };
+      const diagnostics = validateConfigPathsAbsolute(config);
+      expectDiagnostics(diagnostics, {
+        code: "config-path-absolute",
+        message: `Path "my-output" cannot be relative. Use {cwd} or {project-root} to specify what the path should be relative to.`,
+      });
+    });
+
+    it("succeed if using unix absolute path", () => {
+      const config = {
+        ...defaultConfig,
+        projectRoot: "/dev/ws",
+        outputDir: "/my-output",
+      };
+      const diagnostics = validateConfigPathsAbsolute(config);
+      expectDiagnosticEmpty(diagnostics);
+    });
+
+    it("succeed if using windows absolute path", () => {
+      const config = {
+        ...defaultConfig,
+        projectRoot: "/dev/ws",
+        outputDir: "C:/my-output",
+      };
+      const diagnostics = validateConfigPathsAbsolute(config);
+      expectDiagnosticEmpty(diagnostics);
+    });
+  });
```

Comment:
I think we need one test for a circular reference, like:
```yaml
foo:
  default: {bar}
bar:
  default: {foo}
```

Just to exercise the code path for circular reference checking

Reply:
There is already a test for that `"emit diagnostic if variable has circular references"`

Reply:
Missed it, thanks